pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -287,7 +287,8 @@ func resolveDataWithTemplateVars(ctx context.Context, data integration.Data, svc
 		case int, bool:
 
 		default:
-			return data, fmt.Errorf("Unknown type: %T", elem)
+			log.Errorf("Unknown type: %T", elem)
+			continue
```

Comment:
If `null` values are valid and should be accepted,
I would be tempted to explicitly accept specifically `nil` values rather than making no difference between `nil` and any other unexpected type and logging an error in both cases.

```suggestion
		case nil, int, bool:

		default:
			return data, fmt.Errorf("Unknown type: %T", elem)
```

Reply:
👍 Agreed about explicitly handling `nil`. 
However do we really want to return on `default`? Because if we do - we throw away all the substitutions we made. If we `continue` we just ignore the one that failed. 

(either way - since we will now log the error, that will help users understand what is happening, but I'm wondering if we want to be so strict here)

I guess the right thing to do would be to look at the source to `yaml.Unmarshal` and ensure we are handling every possible type it emits so the `default` case never gets hit.